### PR TITLE
[NetAppFiles] Add missing parameter SnapshotName

### PR DESCRIPTION
**Pull request by audunn**
## Description

Adds missing parameter `SnapshotName` to `New-AzNetAppFilesBackup`

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/NetAppFiles/NetAppFiles/Backups/NewNetAppFilesBackup.cs [hunk 1]
```diff
@@ -118,7 +118,7 @@ public class NewAzureRmNetAppFilesBackup : AzureNetAppFilesCmdletBase
         public string VolumeResourceId { get; set; }
 
         [Parameter(
-            Mandatory = true,            
+            Mandatory = false,
```
Comment (dolauli) [inline src/NetAppFiles/NetAppFiles/Backups/NewNetAppFilesBackup.cs:121]: Is this an expected change? PR comments only mentioned adding missing parameter SnapshotName.

File: src/NetAppFiles/NetAppFiles/help/Approve-AzNetAppFilesReplication.md [hunk 2]
```diff
@@ -16,19 +16,19 @@ Approve/Authorize replication connection on the source volume
 ```
 Approve-AzNetAppFilesReplication -ResourceGroupName <String> -AccountName <String> -PoolName <String>
  -Name <String> -DataProtectionVolumeId <String> [-PassThru] [-DefaultProfile <IAzureContextContainer>]
- [-WhatIf] [-Confirm] [<CommonParameters>]
+ [-ProgressAction <ActionPreference>] [-WhatIf] [-Confirm] [<CommonParameters>]
```
Comment (dolauli) [inline src/NetAppFiles/NetAppFiles/help/Approve-AzNetAppFilesReplication.md:19]: Please only update help files for the cmdlet you changed.

**Review by dolauli — CHANGES_REQUESTED** [review]
Some changes are required.

File: src/NetAppFiles/NetAppFiles/Backups/NewNetAppFilesBackup.cs (hunk 1 above)
Comment (audunn) [reply src/NetAppFiles/NetAppFiles/Backups/NewNetAppFilesBackup.cs:121]: Intended, we spoted this during this fix  the `label` parameter is not requred.

File: src/NetAppFiles/NetAppFiles/help/Approve-AzNetAppFilesReplication.md (hunk 2 above)
Comment (audunn) [reply src/NetAppFiles/NetAppFiles/help/Approve-AzNetAppFilesReplication.md:19]: Will do.  
But I would argue if the undrelying cmdlet has not changed it should stay the same. 
Also why is the `Update-MarkdownHelpModule` constantly adding this if needs to be removed manually?

**Comment by audunn** [comment]
@dolauli  PR updated, thanks for your feedback

**Review by dolauli — APPROVED** [review]

**Pull request merged**